packages/remix-dev/vite/plugin.ts

```diff
@@ -537,7 +629,11 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
       serverBundles,
       serverModuleFormat,
       unstable_ssr,
-    };
+    });
+
+    for (let preset of remixUserConfig.presets ?? []) {
+      await preset.remixConfigResolved?.({ remixConfig });
```

Comment:
I'm thinking the args might need to be fleshed out a bit more here. If this hook is designed for config validation, you don't have enough knowledge to provide context for what caused the error, e.g. did the invalid value come from user config, or did it come from another preset, and if so, which one? To aid that last point, we probably also want to ensure that all presets have a `name` property defined.

Reply:
Note, it doesn't have to block this PR, except for adding a required `name` property to presets.

Reply:
The `name` property has been added as a required preset property since we want to ensure all presets have this from day one. We can always add more args to the `remixConfigResolved` hook without it being a breaking change so this is good enough for now.